### DISCUSSION TITLE
"Restore window size when unsnapped" not working correctly

### DISCUSSION THREAD

**Discussion by jeffbdvs** [rxhanson/RectanglePro-Community#566] [Q&A] [answered]
Screen recording link attached. Sorry about the download; file was too big to attach to the post.
Note how anything thrown to the top left corner is unsnapped correctly.
Safari doesn’t unsnap correctly when thrown to the top right corner, but VLC and Photomator do.
Any app thrown to the bottom left or bottom right corners doesn’t unsnap correctly.
Could it be a Settings issue or other user error?
macOS Sonoma 14.7
Mac Studio M2 Ultra
Dual displays both 27”
Rectangle Pro 3.0.35 (176)

[“RectangleProUnsnapProblem” shared from iCloud Drive](https://www.icloud.com/iclouddrive/014VWSm-cm9P-nCTrUxAvWuGw#RectangleProUnsnapProblem)

**Comment by rxhanson** [comment, marked as answer]
Thanks for the details & video. This is a limitation of macOS where it doesn't allow the window to be resized off of the display to the right. Unfortunately, there's not really any good workarounds to fix this as a 3rd party.

**Comment by jeffbdvs** [comment]
Interesting! Thank you for explaining.
Could it/should it be fixed by resizing on mouse up? I don't know all of the downrange implications of doing something like that.

**Reply by rxhanson** [reply]
The main implication of doing this is that it is a different experience (maybe a little confusing) to have it resize on mouse up, and you'd still have to know to move the window far enough away from the side if you wanted it to resize fully (probably not a big deal, though). There's only been one other person to report this on Rectangle, so it hasn't really bubbled up as a priority focus yet.

**Reply by jeffbdvs** [reply]
TL;DR: Thanks, Ryan, for the detailed explanation. As I use Rectangle Pro more and more I'll figure out what works best for me using the tools you've already provided. Everything I've explored so far has been very cool.

 I've always liked my apps to open to the same size and position they were at when I closed them, and I have apps that are only allowed to open in certain spaces (I have 5 across the 2 monitors). So it's not surprising that I want my restored windows to do the same. That probably qualifies as clinical OCD. 😀 Mostly I want Rectangle to help me organize a screen with 6 or 7 apps open that I rotate between while I work in my main app on the other screen. I'm tired of arranging windows so the a corner of this one peeks out from under a corner of the other one, and I put this one here because the window background is a different color than the other 4 on top of it.

Cheers!
